pennylane/_grad.py

```diff
@@ -324,7 +324,15 @@ def _jacobian_function(*args, **kwargs):
                 "If this is unintended, please add trainable parameters via the "
                 "'requires_grad' attribute or 'argnum' keyword."
             )
-        jac = tuple(_jacobian(func, arg)(*args, **kwargs) for arg in _argnum)
+        try:
+            jac = tuple(_jacobian(func, arg)(*args, **kwargs) for arg in _argnum)
+        except TypeError as e:
+            raise ValueError(
```

Comment:
is it worth ensuring `qml.active_return()` before raising this? I'm imagining something like:
1. user gets an unrelated TypeError occurring in jacobian evaluation, sees this custom error
2. user disables return system and tries again
3. still sees this error, can't see what's actually going wrong

I guess if you're here, you're in serious trouble anyway..